src/prefect/automations.py

```diff
@@ -119,6 +156,78 @@ async def update(self: Self):
             )
             await client.update_automation(automation_id=self.id, automation=automation)
 
+    @async_dispatch(aupdate)
+    def update(self: Self):
+        """
+        Updates an existing automation.
+
+        Examples:
+
+
+        ```python
+        auto = Automation.read(id=123)
+        auto.name = "new name"
+        auto.update()
+        ```
+        """
+        assert self.id is not None
+        with get_client(sync_client=True) as client:
+            automation = AutomationCore(
+                **self.model_dump(exclude={"id", "owner_resource"})
+            )
+            client.update_automation(automation_id=self.id, automation=automation)
+
+    @overload
+    @classmethod
+    async def aread(cls, id: UUID, name: Optional[str] = ...) -> Self:
+        ...
+
+    @overload
+    @classmethod
+    async def aread(cls, id: None = None, name: str = ...) -> Optional[Self]:
+        ...
+
+    @classmethod
+    async def aread(
+        cls, id: Optional[UUID] = None, name: Optional[str] = None
+    ) -> Optional[Self]:
+        """
+        Asynchronously read an automation by ID or name.
+
+        Examples:
+
+        ```python
+        automation = await Automation.aread(name="woodchonk")
+        ```
+
+        ```python
+        automation = await Automation.aread(id=UUID("b3514963-02b1-47a5-93d1-6eeb131041cb"))
+        ```
+        """
+        if id and name:
+            raise ValueError("Only one of id or name can be provided")
+        if not id and not name:
+            raise ValueError("One of id or name must be provided")
+        async with get_client() as client:
+            if id:
+                try:
+                    automation = await client.read_automation(automation_id=id)
+                except PrefectHTTPStatusError as exc:
+                    if exc.response.status_code == 404:
+                        raise ValueError(f"Automation with ID {id!r} not found")
+                    raise
+                if automation is None:
+                    raise ValueError(f"Automation with ID {id!r} not found")
+                return cls(**automation.model_dump())
+            else:
+                if TYPE_CHECKING:
+                    assert name is not None
+                automation = await client.read_automations_by_name(name=name)
+                if len(automation) > 0:
+                    return cls(**automation[0].model_dump()) if automation else None
```

Comment:
Seems like we should be consistent and always raise if the automation is not found. That should help simplify the typing.